main.py

```diff
@@ -759,20 +762,47 @@ def handle_link_up(self, event):
 
     # Possibly replace this with interruptions?
     @listen_to(
-        '.*.switch.interface.(link_up|link_down|created|deleted)',
-        pool='dynamic_single'
+        '.*.switch.interface.(link_up|link_down|created|deleted)'
     )
     def on_interface_link_change(self, event: KytosEvent):
         """
-        Handler for interface link_up and link_down events
+        Handler for interface link_up and link_down events.
```

Comment:
Overall loooks good, but let's keep the bare mininmum typically just a single call on any `@listen_to` `on_*` decorated handler since they're tricky to test. So, if you could move/extract all of this logic here into a new `handle_*` just so it's easier and more reliable to unit test that'd be great (basically same guidelines we're using for every `on_*` threaded handler). See what I mean?